### PR TITLE
[FIX] Allow to uninstall addons that extend ir.module.module

### LINKED ISSUE #19158: Cannot uninstall base_import_module
Impacted versions: 10.0 ce/ee

Steps to reproduce:
1. Install base_import_module.
2. Uninstall it.

Current behavior:

```
Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/http.py", line 677, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/http.py", line 935, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/addons/web/controllers/main.py", line 889, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/addons/web/controllers/main.py", line 877, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/addons/base/module/wizard/base_module_upgrade.py", line 76, in upgrade_module
    odoo.modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/modules/registry.py", line 82, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/modules/loading.py", line 393, in load_modules
    Module.browse(modules_to_remove.values()).module_uninstall()
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/addons/base/module/module.py", line 425, in module_uninstall
    self.write({'state': 'uninstalled', 'latest_version': False})
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 3560, in write
    self._write(old_vals)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 3773, in _write
    self.recompute()
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 5314, in recompute
    vals = rec._convert_to_write({n: rec[n] for n in ns})
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 5314, in <dictcomp>
    vals = rec._convert_to_write({n: rec[n] for n in ns})
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 5210, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/fields.py", line 870, in __get__
    self.determine_value(record)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/fields.py", line 956, in determine_value
    self.compute_value(recs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/fields.py", line 936, in compute_value
    self._compute_value(records)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/fields.py", line 927, in _compute_value
    getattr(records, self.compute)()
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/addons/base/module/module.py", line 188, in _get_views
    if module.state not in ('installed', 'to upgrade', 'to remove'):
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/fields.py", line 870, in __get__
    self.determine_value(record)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/fields.py", line 972, in determine_value
    record._prefetch_field(self)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 3066, in _prefetch_field
    result = records.read([f.name for f in fs], load='_classic_write')
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 3010, in read
    self._read_from_database(stored, inherited)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/models.py", line 3134, in _read_from_database
    cr.execute(query_str, params)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/sql_db.py", line 154, in wrapper
    return f(self, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/258853-10-0-47cda5/odoo/sql_db.py", line 231, in execute
    res = self._obj.execute(query, params)
ProgrammingError: column ir_module_module.imported does not exist
LINE 1:  SELECT "ir_module_module"."imported" as "imported","ir_modu...
                ^
```

Expected behavior:
It should be uninstalled, but it seems the uninstallation process tries to read a field that has been removed from `ir.module.module` model. This affects any addon that extends this model.

@Tecnativa

### PR DESCRIPTION
**Pull request by yajo**
Fix #19158.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Is runbot error failing by this PR?

**Comment by Yajo** [comment]
I think the culprit is e88f91ba41d6af028df019893d656a5f50b35d28, let me rebase.

File: odoo/addons/base/module/module.py [hunk 1]
```diff
@@ -422,7 +422,8 @@ def module_uninstall(self):
         """
         modules_to_remove = self.mapped('name')
         self.env['ir.model.data']._module_data_uninstall(modules_to_remove)
-        self.write({'state': 'uninstalled', 'latest_version': False})
+        with self.env.norecompute():
+            self.write({'state': 'uninstalled', 'latest_version': False})
```
Comment (rco-odoo) [inline odoo/addons/base/module/module.py]: By doing this, you discard all recomputations of fields that depend on `state` or `latest_version`.
There are at least three fields that will not be recomputed: `menus_by_module`, `reports_by_module`, `views_by_module` on model `ir.module.module` itself...

Comment (Yajo) [reply odoo/addons/base/module/module.py]: 🤔 I knew this was going to blow up something...

Do you have any other idea?

**Comment by Yenthe666** [comment]
Hey @Yajo,

Thanks a lot for your PR and effort!
In the meanwhile this seems to work fine and it has already been fixed.

Regards,
Yenthe

**Pull request closed without merging**